src/parachain/escrow.ts

```diff
@@ -194,27 +195,30 @@ export class DefaultEscrowAPI implements EscrowAPI {
         let newUserStake = userStake;
         // total_staked_tokens * unlock_heights
         let newTotalStake = totalStake;
-        let newLockDuration = newStakedBalance.endBlock - currentBlockNumber;
+        let newLockDuration = (stakedBalance.endBlock - currentBlockNumber) > 0 ?
+            stakedBalance.endBlock - currentBlockNumber : 0;
         // Case 2: update stake an unlock height
-        if (amountToLock && unlockHeight) {
+        if (amountToLock && blockLockTimeExtension) {
             const monetaryAddedStake = newMonetaryAmount(amountToLock.toBig(), this.governanceCurrency as Currency<GovernanceUnit>);
             newStakedBalance.amount = stakedBalance.amount.add(monetaryAddedStake);
-            newStakedBalance.endBlock = unlockHeight;
+            newStakedBalance.endBlock = stakedBalance.endBlock + blockLockTimeExtension;
 
             newLockDuration = newStakedBalance.endBlock - currentBlockNumber;
             const newStake = amountToLock.toBig().mul(newLockDuration);
             newUserStake = userStake.add(newStake);
             newTotalStake = totalStake.add(newStake);
```

Comment:
I'm not sure I understand this. I thought that extending the lock increases the stake represented by the previously locked amount as well? The lock extension only applies to the newly locked amount in this snippet

Reply:
Greg suggested it would be simpler to compute the vKint amount first and then use `userReward = stake * (totalReward / total_stake)`